bib/cmd/bootc-image-builder/image.go

```diff
@@ -218,6 +218,35 @@ func genPartitionTable(c *ManifestConfig, customizations *blueprint.Customizatio
 	}
 }
 
+// calcRequiredDirectorySizes will calculate the minimum sizes for /
+// for disk customizations. This is only relevant on LVM because on the
+// other types we will expand the "/" to the size of the disk. But on
+// LVM this is not done as we expect people to create VGs with extra space.
+// This should be sufficient because the other cases are:
+//  1. non-LVM is used, "/" is expanded automatically
+//  2. LVM is used ensure "/" is at least rootfsMinSize
+//
+// Note that a custom "/usr" is not supported in image mode so splitting
+// rootfsMinSize between / and /usr is not a concern.
+func calcRequiredDirectorySizes(distCust *blueprint.DiskCustomization, rootfsMinSize uint64) map[string]uint64 {
+	lvmMounts := map[string]uint64{}
+	for _, part := range distCust.Partitions {
+		if part.Type == "lvm" {
+			for _, lv := range part.LogicalVolumes {
+				lvmMounts[lv.Mountpoint] = lv.MinSize
+			}
+		}
+	}
+	// no LVM used, nothing to do
+	if len(lvmMounts) == 0 {
+		return nil
```

Comment:
We need to enforce the root size even on a plain partition, otherwise we get a too small root:

```toml
[[customizations.disk.partitions]]
mountpoint = "/var"
minsize = "20 GiB"
fs_type = "xfs"
```

results into:

```
⏱  Duration: 0s
org.osbuild.mkfs.xfs: f41472c20dedde704b2ba04b2acbab2b270c588ce5d6fcc19c47b0df9ad0f10e {
  "uuid": "00e77a6f-fb67-4525-80a8-111b8b0cc68d",
  "label": "root"
}
device/device (org.osbuild.loopback): loop0 acquired (locked: True)
size 251 of data subvolume is too small, minimum 4096 blocks
```

Reply:
Aha, indeed, I thought we would auto-grow the root for the non-lvm case but I remember we decided we do to support advanced use-cases where special partitions are created on first boot etc. I will make this more generic.

Reply:
This should be fixed now and I added a bunch of tests to ensure this is hopefully well covered now. I can build your example above now.